src/app/auth/resetPassword/index.tsx

```diff
@@ -0,0 +1,250 @@
+import { Link, router } from 'expo-router';
+import React, { useEffect, useRef, useState } from 'react';
+import { Alert, Text, View } from 'react-native';
+import { Icon as RNEIcon } from 'react-native-elements';
+import { ScrollView } from 'react-native-gesture-handler';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import validator from 'validator';
+
+import styles from './styles';
+import Icon from '../../../../assets/icons';
+import StyledButton from '../../../components/StyledButton/StyledButton';
+import UserStringInput from '../../../components/UserStringInput/UserStringInput';
+import colors from '../../../styles/colors';
+import globalStyles from '../../../styles/globalStyles';
+import { useSession } from '../../../utils/AuthContext';
+import supabase from '../../../utils/supabase';
+
+function ResetPasswordScreen() {
+  const { updateUser, signOut } = useSession();
+  const [password, setPassword] = useState('');
+  const [passwordTextHidden, setPasswordTextHidden] = useState(true);
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [confirmPasswordTextHidden, setConfirmPasswordTextHidden] =
+    useState(true);
+  const [loading, setLoading] = useState(false);
+
+  const [hasUppercase, setHasUppercase] = useState(false);
+  const [hasLowercase, setHasLowercase] = useState(false);
+  const [hasNumber, setHasNumber] = useState(false);
+  const [hasLength, setHasLength] = useState(false);
+  const [isDifferent, setIsDifferent] = useState(false);
+  const [isMatching, setIsMatching] = useState(false);
+
+  const checkPassword = (text: string) => {
+    setPassword(text);
+    if (text !== '') {
+      if (text !== text.toLowerCase()) {
```

Comment:
You can simplify these if statements to be `setState(<if statement condition>)`
So like for example this could be `setHasUppercase(text !== text.toLowerCase())`